src/run/run.c

```diff
@@ -1503,15 +1577,24 @@ static int run_context_update(RunContext *c, const char *path) {
         _cleanup_(sd_bus_error_free) sd_bus_error error = SD_BUS_ERROR_NULL;
         int r;
 
-        r = bus_map_all_properties(c->bus,
-                                   "org.freedesktop.systemd1",
-                                   path,
-                                   map,
-                                   BUS_MAP_STRDUP,
-                                   &error,
-                                   NULL,
-                                   c);
+        r = bus_map_all_properties(
+                        c->bus,
+                        "org.freedesktop.systemd1",
+                        c->bus_path,
+                        map,
+                        BUS_MAP_STRDUP,
+                        &error,
+                        NULL,
+                        c);
         if (r < 0) {
+                /* If this is a connection error, then try to reconnect. This might be because the service
+                 * manager is being restarted. Handle this gracefully. */
+                if (sd_bus_error_has_names(&error, SD_BUS_ERROR_NO_REPLY, SD_BUS_ERROR_DISCONNECTED, SD_BUS_ERROR_TIMED_OUT, SD_BUS_ERROR_SERVICE_UNKNOWN, SD_BUS_ERROR_NAME_HAS_NO_OWNER)) {
+                        log_info("Bus call failed to to connection problems. Trying to reconnect...");
+                        run_context_reconnect(c);
+                        return 0;
```

Comment:
I'd like to propagate the error here, and ignore the error on caller side if necessary.

Reply:
so we call this function at three places, and they'd each have to replicate the error handling here, which isn't entirely trivial, because we need to figure out under which errors to retry the connection, and on which errors to fail entirely. Hence, I am not convinced this would be better.

I'll add a comment explaining why we eat up the error here (i.e. because we already handled it by reconnecting)

Reply:
Note, I meant about error propagation from reconnect(). And, yes, of course the error from the previous bus call should be ignored when the reconnection succeeds.